jupyter_server/auth/decorator.py

```diff
@@ -24,7 +27,7 @@ def raise_no_authorizer_warning():
         "https://github.com/jupyter-server/jupyter_server/blob/"
         "653740cbad7ce0c8a8752ce83e4d3c2c754b13cb/jupyter_server/serverapp.py"
         "#L234-L256",
-        # stacklevel=2
+        DeprecationWarning,
```

Comment:
In the [6.x releases of jupyter_client we used `FutureWarning`](https://github.com/jupyter/jupyter_client/blob/6b1aa1cdd39daea559966da75f265ed7a8998602/jupyter_client/manager.py#L415-L416) for deprecation messages as it was performed only once and didn't have the `__main__` constraint that `DeprecationWarning` has.  This might save on the hassle of using a global.

Reply:
Weird, I couldn't get this to display only once, even with `FutureWarning`. I'm not sure if this has something to do warning across separate tornado requests?

Reply:
Nevermind... the issue was using a contextmanager with `FutureWarning`. Removing the `with` block and just warning directly with a `FutureWarning` seemed to accomplish the desired behavior.